haystack/components/evaluators/exact_match_evaluator.py

```diff
@@ -0,0 +1,35 @@
+from itertools import product
+from typing import Any, Dict, List
+
+from haystack import default_from_dict, default_to_dict
+from haystack.core.component import component
+
+
+@component
+class ExactMatchEvaluator:
+    def to_dict(self) -> Dict[str, Any]:
+        return default_to_dict(self)
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "ExactMatchEvaluator":
+        return default_from_dict(cls, data)
+
+    @component.output_types(result=float)
+    def run(self, questions: List[str], ground_truths: List[List[str]], answers: List[List[str]]) -> Dict[str, float]:
+        """
+        `questions` used to extract the answers.
+        `ground_truths` expected answers for each question. We use list of lists as we can have multiple answers candidates.
+        `answers` extracted or generated answers by `questions`. We use list of lists as a question could extract multiple answers.
+
+        All must be the same length.
+        """
+        if not (len(questions) == len(ground_truths) == len(answers)):
+            raise ValueError("The number of questions, ground_truths, and extracted must be the same.")
+
+        matches = 0
+        for truths, extracted in zip(ground_truths, answers):
+            if any(g == e for g, e in product(truths, extracted)):
+                matches += 1
```

Comment:
We can just use `set.intersection` here to calculate the matches and avoid quadratic complexity.